app/models/room.rb

```diff
@@ -48,50 +47,44 @@ def self.updated_room(room, params)
-    company_id_query = company_ids.map{ |id| "company_id = #{id}" }.join(" OR ")
-    company_id_query.present? ? company_id_query.prepend(" AND (") << ")" : company_id_query = " AND company_id = 0"
-
-    name = params.fetch("name", "")
-    max_occupancy = params.fetch("max_occupancy", "").empty? ? "-1" : params["max_occupancy"]
-    room_number = params.fetch("room_number", "").empty? ? "%%" : params["room_number"]
-    location = params.fetch("location", "").empty? ? "%%" : params["location"]
-
-    where_query = "lower(name) LIKE lower('%#{name}%') AND
-                   max_occupancy >= #{max_occupancy} AND
-                   CAST(room_number AS TEXT) LIKE '#{room_number}' AND
-                   lower(location) LIKE lower('#{location}')" +
-                   company_id_query
-
-    Room.where(where_query)
+  def self.company_rooms(company)
+    self.where(company_id: company).map { |room| [room.name, room.id] }
   end
 
-  def self.sort_with(params)
-    sort_by   = params.fetch("sort_by", "name||").split("||")
-    sort_dir  = params.fetch("sort_dir", "ASC||").split("||")
-    sort_hash = Hash[sort_by.zip(sort_dir)]
+  def self.search_for(params)
+    search = params[:search].downcase
+    max_occupancy = params["max_occupancy"].empty? ? "-1" : params["max_occupancy"]
+    room_number = params["room_number"].empty? ? "%%" : params["room_number"]
 
-    order_query = sort_hash.map { |sort_by, sort_dir| "#{sort_by} #{sort_dir}" }.join(", ")
-    self.order(order_query)
-  end
+    amenities = Amenity.where("lower(perk) LIKE lower('%#{search}%')")
+    room_ids = amenities.map{ |amenity| amenity.rooms.pluck(:id) }.flatten.uniq
 
-  def self.company_rooms(company)
-    self.where(company_id: company).map { |room| [room.name, room.id] }
+    amenity_query = room_ids.map{ |id| "id = #{id}" }.join(" OR ")
+    amenity_query.empty? ? amenity_query = " OR id = 0)" : amenity_query.prepend(" OR (") << "))"
+    search_query = "max_occupancy >= #{max_occupancy}
+                    AND CAST(room_number AS TEXT) LIKE '#{room_number}'
+                    AND (lower(name) LIKE '%#{search}%'
+                    OR lower(location) LIKE ('%#{search}%')" + amenity_query
+    self.where(search_query)
   end
 
-  def self.search_for(search)
-    if search.to_i == 0
-      max_occupancy = 1000
-    else
-      max_occupancy = search.to_i
+  def get_next_meeting_details
+    time = "N/A"
+    available = "yes"
+    next_meeting = nil
+    now = Time.now
+    unless meetings.none?
+      next_start_time = self.meetings.first.start_time
+      self.meetings.each do |meeting|
+        start_time = meeting.start_time
+        available = "no" if start_time <= now && meeting.end_time >= now
+        if start_time > now && start_time <= next_start_time
+          next_start_time = start_time
+          time = start_time
+          next_meeting = meeting
+        end
+      end
```

Comment:
This method is rather impenetrable. I cannot even tell what the return value is from this. 

Is there any way this could be broken into smaller, more readable methods that better convey their intent?